Merge from upstream v13

Alchez opened the pull request:
**Changelogs:**

- https://github.com/frappe/frappe/releases/tag/v13.31.0
- https://github.com/frappe/frappe/releases/tag/v13.32.0
- https://github.com/frappe/frappe/releases/tag/v13.33.0
  - https://github.com/frappe/frappe/releases/tag/v13.33.1
- https://github.com/frappe/frappe/releases/tag/v13.34.0

Commit:
bug: db_query.py

The object 'meta' does not always have the attribute 'is_submittable.'
This extra syntax checks for existence first.

(cherry picked from commit d8113698d1322304dd70c6797485fe347507a4c0)

mituldavid committed:
fix: Make images in comments private

(cherry picked from commit d254ca2f505833c6d74a5302c283b574f4d8ae9d)

# Conflicts:
#	frappe/core/doctype/file/file.py

mituldavid committed:
fix: Optional parameter for setting extracted image as private

(cherry picked from commit 05cb2c16cd037a9b6110a45295a46bfe2c07295b)

# Conflicts:
#	frappe/desk/form/utils.py

mituldavid committed:
fix: Private images in comments are not visible

shariquerik committed:
fix: prioritize report custom roles

(cherry picked from commit 5bd3fd8d6515b45df11295dad8eaed64fffa3e56)

shariquerik committed:
test: added unit test for report custom role permissions

(cherry picked from commit a3d9d8028f03009ecc6ceab490b4967d310232e0)

shariquerik committed:
chore: removed unused variable & added nosemgrep on db commit

(cherry picked from commit 95921c870762a9ef7fbff0a7f9f184df2e223fd5)

barredterra committed:
feat: delete User Permission when User is deleted

(cherry picked from commit a0e8983199cf9c4f176377458025dafb6929b0e6)

mergify[bot] committed:
test: flaky test due to stale translation cache (backport #16985) (#16993)

Co-authored-by: Ankush Menat <ankush@frappe.io>

shariquerik committed:
fix: Strip all spacing characters from Message-ID & In-Reply-To (backport #16999) (#17004)

finbyz committed:
Update email_account.py

If reference_doctype and reference_name is getting None then Communication will not link with append_to DocType.

gavindsouza committed:
fix: Don't update modified time on set_notifications_as_unseen

(cherry picked from commit 1df1bc5232e9dd7f459deb5e6febaefb84920af4)

gavindsouza committed:
fix: Update Notification Settings via doc.save

toggle_notifications is called via User doc. Changes made through
another doc should leave some trace. Using doc.save in order to maintain
versions.

(cherry picked from commit 3c7dd75047ac190c1a00b2953c40cb9fd98ddf2b)

mergify[bot] committed:
feat(website-settings): Configurable Splash Image (backport #17006) (#17029)

* feat(website-settings): Configurable Splash Image

Configurable Splash Image option which will override app's splash_image
hook set.

(cherry picked from commit f416e2a1d2ddeaa314cc6b2ec31c1e5ec727e1dc)

* fix: Add splash_image to context if exists

(cherry picked from commit 9bc85402f67044ebce46ebd1ba1391053ec9837d)

* fix: Resolve conflicts + simplify logic

Co-authored-by: gavin <gavin18d@gmail.com>

surajshetty3416 committed:
feat: Auto-expire web view link key

surajshetty3416 committed:
fix: Add a patch to update document_share_key_expiry defaults

surajshetty3416 committed:
fix: Respond with http_status_code set in context

surajshetty3416 committed:
test: Case to validate Web Link authentication

FHenry committed:
fix: translation context for print format align values (#17026)

(cherry picked from commit 1826c9e9afd6311dc92e830a700a52f9fb487540)

# Conflicts:
#	frappe/public/js/print_format_builder/PrintFormatControls.vue

shariquerik committed:
fix: deleted PrintFormatControls.vue

mergify[bot] committed:
Fix: Delete Share in Frontend when read is de-selected (backport #17020) (#17033)

Co-authored-by: Tom-Finke <tom.finke@webterra.de>

deepeshgarg007 committed:
Merge pull request #17037 from frappe/version-13-hotfix

chore: Weekly release for version-13

frappe-bot committed:
chore(release): Bumped to Version 13.31.0

# [13.31.0](https://github.com/frappe/frappe/compare/v13.30.0...v13.31.0) (2022-05-31)

### Bug Fixes

* Strip all spacing characters from Message-ID & In-Reply-To (backport [#16999](https://github.com/frappe/frappe/issues/16999)) ([#17004](https://github.com/frappe/frappe/issues/17004)) ([6bc719d](https://github.com/frappe/frappe/commit/6bc719d5960d714cdf3a928fa6d74ad9f9fb2e4c))

### Features

* **website-settings:** Configurable Splash Image (backport [#17006](https://github.com/frappe/frappe/issues/17006)) ([#17029](https://github.com/frappe/frappe/issues/17029)) ([2c82a0f](https://github.com/frappe/frappe/commit/2c82a0fd6f45f96448e2cac1c173939f52c9b4e9))

surajshetty3416 committed:
perf: Simplify code to avoid unnecessary DB calls

surajshetty3416 committed:
refactor: Update exception names

surajshetty3416 committed:
Merge branch 'version-13-hotfix' into mergify/bp/version-13-hotfix/pr-17026

surajshetty3416 committed:
Merge pull request #17049 from frappe/mergify/bp/version-13-hotfix/pr-16976

surajshetty3416 committed:
Merge pull request #17036 from frappe/mergify/bp/version-13-hotfix/pr-17026

surajshetty3416 committed:
Merge pull request #17050 from frappe/mergify/bp/version-13-hotfix/pr-16981

mergify[bot] committed:
perf: store compact JSON in version diffs (backport #17042) (#17044)

perf: store compact JSON in version diffs (#17042)

(cherry picked from commit f8f29e284fd885b7ff23a050a24894b5a17bd711)

Co-authored-by: Ankush Menat <ankush@frappe.io>

surajshetty3416 committed:
fix: Add "Document Share Key" to `doctypes_to_skip` in `delete_doc.py`

- To avoid error while deleting linked document.

surajshetty3416 committed:
fix: Randomize key length to make it harder to brute force

surajshetty3416 committed:
Merge pull request #17035 from surajshetty3416/document-share-key-with-expiry

mergify[bot] committed:
fix: update fr translation (backport #17070) (#17073)

fix: update fr translation (#17070)

(cherry picked from commit e0c25d2b2c19fe79f7c7848e7307d90a5f27c68a)

Co-authored-by: HENRY Florian <florian.henry@open-concept.pro>

Commit:
Merge remote-tracking branch 'upstream/version-13' into merge-v13

ghost commented:
### Tests

* flaky test due to stale translation cache (backport #16985) (#16993) ([942551e](https://github.com/ParsimonyGit/frappe/commit/942551e3eeff4ac8b21687db0a8fbb458c24089f))
* Case to validate Web Link authentication ([db60ae4](https://github.com/ParsimonyGit/frappe/commit/db60ae4fa89a7139774031094718f5ae3e3f0058))
* added unit test for report custom role permissions ([8614537](https://github.com/ParsimonyGit/frappe/commit/8614537ca01c16b340a44ffe6855b00b87943f19))


### Bug Fixes

* Strip all spacing characters from Message-ID & In-Reply-To (backport #16999) (#17004) ([6bc719d](https://github.com/ParsimonyGit/frappe/commit/6bc719d5960d714cdf3a928fa6d74ad9f9fb2e4c))
* Add a patch to update document_share_key_expiry defaults ([5ededb0](https://github.com/ParsimonyGit/frappe/commit/5ededb0361976275fb8faf49875d4a4d4389333f))
* Respond with http_status_code set in context ([b4c7fa0](https://github.com/ParsimonyGit/frappe/commit/b4c7fa0bf9c0f1a0da4b91654c06f1c4493e6dac))
* translation context for print format align values (#17026) ([0a02939](https://github.com/ParsimonyGit/frappe/commit/0a02939984edccbe80a0bb3c5bb11da63cc59551))
* deleted PrintFormatControls.vue ([5fa784c](https://github.com/ParsimonyGit/frappe/commit/5fa784cb3b970a333e4ef81c735c9cfd2b9a5198))
* Delete Share in Frontend when read is de-selected (backport #17020) (#17033) ([460bc44](https://github.com/ParsimonyGit/frappe/commit/460bc44a54bda175850929c426e74f6e62e0b191))
* prioritize report custom roles ([b3bef50](https://github.com/ParsimonyGit/frappe/commit/b3bef505f3c1c308683228ff391611d2832d532f))
* Add "Document Share Key" to `doctypes_to_skip` in `delete_doc.py` ([75b79ee](https://github.com/ParsimonyGit/frappe/commit/75b79eefa637b1accc0afbdd6456a5816a1df15c))
* Randomize key length to make it harder to brute force ([1da96ea](https://github.com/ParsimonyGit/frappe/commit/1da96eaeeefd41c3ec648fac72883c5f2d223d21))
* update fr translation (backport #17070) (#17073) ([0ed5fe1](https://github.com/ParsimonyGit/frappe/commit/0ed5fe19919fd95cd51afc04419b2c4c73c7f8fc))


### Features

*  **website-settings:** Configurable Splash Image (backport #17006) (#17029) ([2c82a0f](https://github.com/ParsimonyGit/frappe/commit/2c82a0fd6f45f96448e2cac1c173939f52c9b4e9))
* Auto-expire web view link key ([c1fee82](https://github.com/ParsimonyGit/frappe/commit/c1fee82c81ef143ecc0cf1cb3fd3f816c2dbbd95))
* delete User Permission when User is deleted ([cb0f323](https://github.com/ParsimonyGit/frappe/commit/cb0f3236f958332e1bf104132064364f4ec03995))


### Chore

*  **release:** Bumped to Version 13.31.0 ([6d3f126](https://github.com/ParsimonyGit/frappe/commit/6d3f1266000b0e86322f1f9d71f40e9f50dc5e40))
* removed unused variable & added nosemgrep on db commit ([491eaa5](https://github.com/ParsimonyGit/frappe/commit/491eaa590bb71e71c411189c7b8125afd3f21d8b))


### Performance Improvements

* Simplify code to avoid unnecessary DB calls ([ad6aafb](https://github.com/ParsimonyGit/frappe/commit/ad6aafb0eb2d3b7200de74b9617fb5899fa4ab42))
* store compact JSON in version diffs (backport #17042) (#17044) ([4cda7af](https://github.com/ParsimonyGit/frappe/commit/4cda7af4cd4b52624dacc6d1d78a58df0d04bf0b))


### Code Refactoring

* Update exception names ([3d6f6d1](https://github.com/ParsimonyGit/frappe/commit/3d6f6d10ea0857b1f90781aebbe354a99cc884f3))


### Reverts

* "fix: mention only users that can be read" (backport #17074) (#17076) ([09b960c](https://github.com/ParsimonyGit/frappe/commit/09b960cfd07ffeea2822aec4ff7e6e3fd7cc92d6))


### Contributors

<a href="https://github.com/mergify">mergify[bot]</a>, <a href="https://github.com/shariquerik">shariquerik</a>, <a href="https://github.com/finbyz">finbyz</a>, <a href="https://github.com/surajshetty3416">surajshetty3416</a>, <a href="https://github.com/FHenry">FHenry</a>, <a href="https://github.com/deepeshgarg007">deepeshgarg007</a>, <a href="https://github.com/frappe-bot">frappe-bot</a>, <a href="https://github.com/barredterra">barredterra</a>, <a href="https://github.com/Alchez">Alchez</a>

<details>
<summary>Commit-Lint commands</summary>
<br />

You can trigger Commit-Lint actions by commenting on this PR:
- `@Commit-Lint merge patch` will merge dependabot PR on "patch" versions (X.X.Y - Y change)
- `@Commit-Lint merge minor` will merge dependabot PR on "minor" versions (X.Y.Y - Y change)
- `@Commit-Lint merge major` will merge dependabot PR on "major" versions (Y.Y.Y - Y change)
- `@Commit-Lint merge disable` will desactivate merge dependabot PR
- `@Commit-Lint review` will approve dependabot PR
- `@Commit-Lint stop review` will stop approve dependabot PR



</details>

mergify[bot] committed:
revert: "fix: mention only users that can be read" (backport #17074) (#17076)

Revert "fix: mention only users that can be read (#16978)"

This reverts commit 576af52ec6039beb344388030112ad3de6f6d2d9.

(cherry picked from commit d9900d474226e90e2fbc20afa694a15db4bab4ab)

Co-authored-by: Ritwik Puri <ritwikpuri5678@gmail.com>

mergify[bot] committed:
revert: "fix: mention only users that can be read" (backport #17074)

mergify[bot] committed:
ci: update apt cache (backport #17057) (#17077)

* ci: update apt cache (#17057)

(cherry picked from commit 9137a597ebed0fb81f39405deb54b49dce1bce11)

# Conflicts:
#	.github/helper/install.sh

* chore: conflicts

* chore: remove mariadb client install 

not required on ubuntu 18

Co-authored-by: Ankush Menat <ankush@frappe.io>

ankush committed:
ci: disable parallel tests on v13 (#17079)

- After removing coverage reporting, the tests run much faster
- 5-6 min for setup and 2-3 minute for test on each container is just
  not optimum / waste of resources.

mergify[bot] committed:
fix: duplicate buttons on desk (backport #16941) (#16952)

* fix: duplicate buttons on desk (#16941)

(cherry picked from commit fe8520a2bdd18dda3c1e23577a231124e8ac5a4d)

# Conflicts:
#	cypress/integration/folder_navigation.js
#	frappe/public/js/frappe/ui/page.js

* fix: resolved conflicts

* fix: es6 compatibility

* chore: remove conflicting files

Co-authored-by: Ankush Menat <ankush@frappe.io>

shariquerik committed:
fix(minor): Query Report broken UI (#17088)

FHenry committed:
fix: update fr translation (#17090)

(cherry picked from commit f4ccac24b7a37e15a612882d751d0333dc2a566d)

deepeshgarg007 committed:
Merge pull request #17091 from frappe/version-13-hotfix

chore: weekly version 13 release

frappe-bot committed:
chore(release): Bumped to Version 13.32.0

# [13.32.0](https://github.com/frappe/frappe/compare/v13.31.0...v13.32.0) (2022-06-07)

### Bug Fixes

* Add "Document Share Key" to `doctypes_to_skip` in `delete_doc.py` ([75b79ee](https://github.com/frappe/frappe/commit/75b79eefa637b1accc0afbdd6456a5816a1df15c))
* Add a patch to update document_share_key_expiry defaults ([5ededb0](https://github.com/frappe/frappe/commit/5ededb0361976275fb8faf49875d4a4d4389333f))
* deleted PrintFormatControls.vue ([5fa784c](https://github.com/frappe/frappe/commit/5fa784cb3b970a333e4ef81c735c9cfd2b9a5198))
* duplicate buttons on desk (backport [#16941](https://github.com/frappe/frappe/issues/16941)) ([#16952](https://github.com/frappe/frappe/issues/16952)) ([b637d76](https://github.com/frappe/frappe/commit/b637d76654f8313ab2dd0f972fa4bf578ce18b1e))
* **minor:** Query Report broken UI ([#17088](https://github.com/frappe/frappe/issues/17088)) ([3bf154c](https://github.com/frappe/frappe/commit/3bf154c7ce473a6b1ad9e536242e3071fa4d9466))
* prioritize report custom roles ([b3bef50](https://github.com/frappe/frappe/commit/b3bef505f3c1c308683228ff391611d2832d532f))
* Randomize key length to make it harder to brute force ([1da96ea](https://github.com/frappe/frappe/commit/1da96eaeeefd41c3ec648fac72883c5f2d223d21))
* Respond with http_status_code set in context ([b4c7fa0](https://github.com/frappe/frappe/commit/b4c7fa0bf9c0f1a0da4b91654c06f1c4493e6dac))
* translation context for print format align values ([#17026](https://github.com/frappe/frappe/issues/17026)) ([0a02939](https://github.com/frappe/frappe/commit/0a02939984edccbe80a0bb3c5bb11da63cc59551))
* update fr translation (backport [#17070](https://github.com/frappe/frappe/issues/17070)) ([#17073](https://github.com/frappe/frappe/issues/17073)) ([0ed5fe1](https://github.com/frappe/frappe/commit/0ed5fe19919fd95cd51afc04419b2c4c73c7f8fc))

### Features

* Auto-expire web view link key ([c1fee82](https://github.com/frappe/frappe/commit/c1fee82c81ef143ecc0cf1cb3fd3f816c2dbbd95))
* delete User Permission when User is deleted ([cb0f323](https://github.com/frappe/frappe/commit/cb0f3236f958332e1bf104132064364f4ec03995))

### Performance Improvements

* Simplify code to avoid unnecessary DB calls ([ad6aafb](https://github.com/frappe/frappe/commit/ad6aafb0eb2d3b7200de74b9617fb5899fa4ab42))
* store compact JSON in version diffs (backport [#17042](https://github.com/frappe/frappe/issues/17042)) ([#17044](https://github.com/frappe/frappe/issues/17044)) ([4cda7af](https://github.com/frappe/frappe/commit/4cda7af4cd4b52624dacc6d1d78a58df0d04bf0b))

### Reverts

* "fix: mention only users that can be read" (backport [#17074](https://github.com/frappe/frappe/issues/17074)) ([67a6074](https://github.com/frappe/frappe/commit/67a607410629f7ee136fa9e15dfc55da0ade876e))
* "fix: mention only users that can be read" (backport [#17074](https://github.com/frappe/frappe/issues/17074)) ([#17076](https://github.com/frappe/frappe/issues/17076)) ([09b960c](https://github.com/frappe/frappe/commit/09b960cfd07ffeea2822aec4ff7e6e3fd7cc92d6)), closes [#16978](https://github.com/frappe/frappe/issues/16978)

surajshetty3416 committed:
fix: Reset role editor while loading new user form

(cherry picked from commit 0c28820bcce517f973d61d564530da1738b1ca14)

surajshetty3416 committed:
fix: Avoid unnecessary "Not Found" error message

(cherry picked from commit fca0853ebd4827b4bb6d462fe2229ea0f277e61a)

surajshetty3416 committed:
Merge pull request #17101 from frappe/mergify/bp/version-13-hotfix/pr-17090

surajshetty3416 committed:
Merge pull request #17105 from frappe/mergify/bp/version-13-hotfix/pr-17024

surajshetty3416 committed:
Merge pull request #17106 from frappe/mergify/bp/version-13-hotfix/pr-17104

fix: Reset role editor while loading a new user form (backport #17104)

mergify[bot] committed:
fix: infinite loop graphic bar (backport #17097) (#17109)

fix: infinite loop graphic bar

When Period is Yearly and Date Range is Last Year in Graphic Bar

(cherry picked from commit 4850c0d4fe2f189ad447c42037bf5da5e8d0b9fa)

refactor: simplify timegrain conditions

All special handling for timegrains is same as general handling, no idea why it's done
like this.

(cherry picked from commit 59a9be4caf4ac20a715a510d8305975820436ea0)

test: basic setup tests

Co-authored-by: mmdanny89 <66078599+mmdanny89@users.noreply.github.com>

Commit:
Merge remote-tracking branch 'upstream/version-13' into merge-v13

RJPvT committed:
fix: no add/change image-field if user is not allowed

no add/change image-field (dropdown) if user is not allowed to change

(cherry picked from commit 4af5006470739a11ddcf3141c95ae6fb157d6825)

RJPvT committed:
Update user_image.js

(cherry picked from commit d9d3b1421e5516fa8471e5afedebed99ed3bd095)

mergify[bot] committed:
fix: print & breadcrumbs for forward slash (backport #17095) (#17110)

Co-authored-by: Maharshi Patel <39730881+maharshivpatel@users.noreply.github.com>

mergify[bot] committed:
fix: Switch to form view from print preview page (backport #17087) (#17112)

Co-authored-by: Vishal Dhayagude <vishdha@users.noreply.github.com>

mergify[bot] committed:
fix: added text editor dialog on hide action (backport #17117) (#17125)

Co-authored-by: Shariq Ansari <30859809+shariquerik@users.noreply.github.com>

mergify[bot] committed:
feat(UI): Add beforeunload Listener for 'Leave Site' Dialog to dirty forms (backport #17059) (#17129)

Co-authored-by: Tom-Finke <tom.s.h.finke@gmail.com>

netchampfaris committed:
fix: Allow to set empty values in Bulk Edit

- minor UX fixes

(cherry picked from commit 076816a7d7d7541b66cc7c3d6a977256b77642dd)

sagarvora committed:
fix: return found items from duplicate btn check (#17147)

mergify[bot] committed:
fix: return found items from duplicate btn check (backport #17147) (#17148)

fix: return found items from duplicate btn check (#17147)

(cherry picked from commit a8ff3653ebfff24d738edd1fa8b7f8a92b7b0863)

Co-authored-by: Sagar Vora <sagar@resilient.tech>

frappe-bot committed:
chore(release): Bumped to Version 13.32.1

## [13.32.1](https://github.com/frappe/frappe/compare/v13.32.0...v13.32.1) (2022-06-11)

### Bug Fixes

* return found items from duplicate btn check (backport [#17147](https://github.com/frappe/frappe/issues/17147)) ([#17148](https://github.com/frappe/frappe/issues/17148)) ([a9def42](https://github.com/frappe/frappe/commit/a9def42c3936739550b56b5837d9c610bf10f59c))

FHenry committed:
fix: JS error in page.js when the is a quote in button translation

(cherry picked from commit 89889753e4dadab618bdbf8101823e8187cc995d)

ankush committed:
test: button with single quote in label

(cherry picked from commit 03f48580ef1baa91b3d1bd7fdb90e20fa3f9adb4)

surajshetty3416 committed:
Merge pull request #17155 from frappe/mergify/bp/version-13-hotfix/pr-17152

fix: JS error in page.js when the is a quote in button translation (backport #17152)

surajshetty3416 committed:
Merge pull request #17138 from frappe/mergify/bp/version-13-hotfix/pr-13441

fix: check is_submittable attr exists in meta before accessing (backport #13441)

surajshetty3416 committed:
Merge pull request #17022 from finbyz/patch-5

surajshetty3416 committed:
Merge pull request #17157 from frappe/mergify/bp/version-13-hotfix/pr-17136

fix: Allow to set empty values in Bulk Edit (backport #17136)

mergify[bot] committed:
perf: disable creating version for new docs (backport #17153) (#17158)

* perf: disable creating version for new docs

Each new doc inserts a version, this contains nothing but creator and
creation time.. which is already immutable information on the original
document.

This was added for cases like data import to track from where document
got created, ref: https://github.com/frappe/frappe/commit/b7dfe7969dd6b2851844b1e9c090cbb6447748de

Fix: only add a version on creation IF creation info is present on flags
(cherry picked from commit 21442f5cbaa5c7b1a5a0aa1c8a376ddbc6d04ad6)

* refactor: use comprehension instead of filter

Filter objects have caused troubles in past because they only allow
iterating over them once.

Co-authored-by: Ankush Menat <ankush@frappe.io>

surajshetty3416 committed:
fix: Handle case where document title can be NONE

(cherry picked from commit 9b67fc24bc290789158f37a8f2ce10b505878792)

surajshetty3416 committed:
fix: Handle case where document title can be NONE

surajshetty3416 committed:
fix: `doc.get_title` should return empty string if title is not set

(cherry picked from commit b244c9148192362de43aa483b328709745f3be82)

surajshetty3416 committed:
fix: `doc.get_title` should return empty string if title is not set

surajshetty3416 committed:
Merge pull request #17161 from surajshetty3416/fix-printview

surajshetty3416 committed:
Merge pull request #17162 from frappe/mergify/bp/version-13/pr-17161

fix: Handle case where document title can be NONE (backport #17161)

frappe-bot committed:
chore(release): Bumped to Version 13.32.2

## [13.32.2](https://github.com/frappe/frappe/compare/v13.32.1...v13.32.2) (2022-06-13)

### Bug Fixes

* `doc.get_title` should return empty string if title is not set ([3e97521](https://github.com/frappe/frappe/commit/3e97521a7a3e835bed5b1988e3dea163d3ec306e))
* Handle case where document title can be NONE ([5a27da5](https://github.com/frappe/frappe/commit/5a27da53195d2dc925da154f92937f1ab77a3bd2))

Abhirup-99 committed:
fix: Update label for editing in full page and add custom logic for rendering it. (#17149)

(cherry picked from commit 8c8249894beb0b6707954485e2b3947192e8025b)

deepeshgarg007 committed:
Merge pull request #17172 from frappe/version-13-hotfix

chore: weekly version-13 release

frappe-bot committed:
chore(release): Bumped to Version 13.33.0

# [13.33.0](https://github.com/frappe/frappe/compare/v13.32.2...v13.33.0) (2022-06-14)

### Bug Fixes

* `doc.get_title` should return empty string if title is not set ([b244c91](https://github.com/frappe/frappe/commit/b244c9148192362de43aa483b328709745f3be82))
* added text editor dialog on hide action (backport [#17117](https://github.com/frappe/frappe/issues/17117)) ([#17125](https://github.com/frappe/frappe/issues/17125)) ([3c56747](https://github.com/frappe/frappe/commit/3c56747ea4b261983395debf6afdafaf9a869567))
* Allow to set empty values in Bulk Edit ([e7e5a4a](https://github.com/frappe/frappe/commit/e7e5a4a114a61bfc95088946ec4b646d3f4806ff))
* Avoid unnecessary "Not Found" error message ([4536b4d](https://github.com/frappe/frappe/commit/4536b4d6c1d6ff173a144e88b466f47efc49bdd1))
* Don't update modified time on set_notifications_as_unseen ([b5ab97d](https://github.com/frappe/frappe/commit/b5ab97d283fcd045076c64db25437ab432d76fb5))
* Handle case where document title can be NONE ([9b67fc2](https://github.com/frappe/frappe/commit/9b67fc24bc290789158f37a8f2ce10b505878792))
* infinite loop graphic bar (backport [#17097](https://github.com/frappe/frappe/issues/17097)) ([#17109](https://github.com/frappe/frappe/issues/17109)) ([90c8e9f](https://github.com/frappe/frappe/commit/90c8e9f29240ae696b10e6240ade91174fd35ce0))
* JS error in page.js when the is a quote in button translation ([ab657b4](https://github.com/frappe/frappe/commit/ab657b452b145438289471a1ab4383e6b27c4e35))
* print & breadcrumbs for forward slash (backport [#17095](https://github.com/frappe/frappe/issues/17095)) ([#17110](https://github.com/frappe/frappe/issues/17110)) ([cc4c570](https://github.com/frappe/frappe/commit/cc4c5701e89549e2d59b53ef5c07ff5dc2dce23f))
* Reset role editor while loading new user form ([64744f9](https://github.com/frappe/frappe/commit/64744f9e5d083b1fcbec92c6e88e900d44feb4e8))
* return found items from duplicate btn check ([#17147](https://github.com/frappe/frappe/issues/17147)) ([a8ff365](https://github.com/frappe/frappe/commit/a8ff3653ebfff24d738edd1fa8b7f8a92b7b0863))
* Switch to form view from print preview page (backport [#17087](https://github.com/frappe/frappe/issues/17087)) ([#17112](https://github.com/frappe/frappe/issues/17112)) ([20e0c4e](https://github.com/frappe/frappe/commit/20e0c4e217b01e43b3d8b9096bc6e0d8bd4c58eb))
* update fr translation ([#17090](https://github.com/frappe/frappe/issues/17090)) ([19bae66](https://github.com/frappe/frappe/commit/19bae66783969a2cb729a8e5ee5d93249c17feff))
* Update Notification Settings via doc.save ([85d0a70](https://github.com/frappe/frappe/commit/85d0a702ce10e1d0f8bf6a4ef8819b95eb2fc014))

### Features

* **UI:** Add beforeunload Listener for 'Leave Site' Dialog to dirty forms (backport [#17059](https://github.com/frappe/frappe/issues/17059)) ([#17129](https://github.com/frappe/frappe/issues/17129)) ([fa63d46](https://github.com/frappe/frappe/commit/fa63d46539b1faa0b8f1ca964600fd03f40293ad))

### Performance Improvements

* disable creating version for new docs (backport [#17153](https://github.com/frappe/frappe/issues/17153)) ([#17158](https://github.com/frappe/frappe/issues/17158)) ([192e31e](https://github.com/frappe/frappe/commit/192e31e406988ee2f17588f506938f7b39b6882b))

Commit:
Merge remote-tracking branch 'upstream/version-13' into merge-v13

ruchamahabal committed:
fix(Event Streaming): set remote docname and site name before inserting mapped doc

- needed to differentiate a remote doc at the time of insertion

(cherry picked from commit 4c965d54df61b02127e6f736b28d132594e88033)

ruchamahabal committed:
fix: use the correct dt name while populating data in mapping table

(cherry picked from commit a5c04561d0f573efa619d2f108bc1123bb01b883)

mergify[bot] committed:
fix: allow All to select a User (backport #16983) (#17178)

* fix: allow All to select a User

(cherry picked from commit a7c706672f4eb28bdaecb6109e042070db1da188)

* test: user permissions affecting User

(cherry picked from commit cdc850f12eb52214fb7d4d6d37b3f5fbc75ee08c)

# Conflicts:
#	frappe/tests/test_permissions.py

* fix: resolve merge conflicts

Co-authored-by: barredterra <14891507+barredterra@users.noreply.github.com>

surajshetty3416 committed:
Merge pull request #17168 from frappe/mergify/bp/version-13-hotfix/pr-17149

surajshetty3416 committed:
fix: Add image_field check

(cherry picked from commit e48f6620646e213526b7ba5614779558e729c638)

surajshetty3416 committed:
Merge pull request #17190 from frappe/mergify/copy/version-13-hotfix/pr-17116

surajshetty3416 committed:
fix: Resolve conflicts

mergify[bot] committed:
fix(db_query): Remove wanton whitespace split check (backport #17180) (#17195)

* fix: Remove unwanted blacklist over fields

A field like 'count(`tabBOM Update Log`.name) as total_count' split by
whitespace loses it's meaning. Tried splitting it meaningfully but
didn't get the point of this tbh and stopped. I'm not sure what the code
before was trying to do and with what set of inputs.

Imagine the following fields: [count(`tabBOM Update Log`.name) as
total_count, `tabBOM Update Log`.name as update_name, `tabBOM Update
Log`.name, `tabBOM Update as Log`.name, tabBOM.name, name], I couldn't
see what the previous check was trying to protect - hence, didn't add
any equivalent functionality.

(cherry picked from commit 60ec324956164eaa95709dde48226a19b2b4035b)

* perf: Limit re internal cache to avoid caching patterns twice

(cherry picked from commit 678eebe4fb8531413b88865f3e1bea162758910e)

* perf(db_query): Maintain compiled pattern globally

(cherry picked from commit 9b4db43b84ead21a135fa0ce29e810306c01f9b5)

* chore: Resolve porting conflicts

Co-authored-by: Gavin D'souza <gavin18d@gmail.com>

surajshetty3416 committed:
Merge pull request #17198 from frappe/mergify/bp/version-13-hotfix/pr-14371

fix: Make images in comments private (backport #14371)

ankush committed:
Revert "fix: allow All to select a User (backport #16983)" (#17199)

Revert "fix: allow All to select a User (backport #16983) (#17178)"

This reverts commit 210ad01177a5b90ec8ce9e8500b66c9a4748368f.

mergify[bot] committed:
fix: format date as per system settings for dashboard charts (backport #17151) (#17203)

* fix: format date as per system settings for dashboard charts

(cherry picked from commit 536f88aafc6892250a524140ee773295cfa8c67d)

* test: test_user_date_label_dashboard_chart

(cherry picked from commit 57a4d590d0e80dd5a4a0e1df05fe72dfc84bfe64)

* test: Fix dashboard chart test cases

(cherry picked from commit 77a6c103ec6c55b5964835a85fad158ed3d2be4b)

* fix: Resolve conflicts

Co-authored-by: phot0n <ritwikpuri5678@gmail.com>
Co-authored-by: Suraj Shetty <surajshetty3416@gmail.com>

ruchamahabal committed:
Merge pull request #17216 from frappe/mergify/bp/version-13-hotfix/pr-17177

fix(Event Streaming): Set remote docname and site name before mapped doc insertion + fix mapping setup (backport #17177)

mergify[bot] committed:
fix: number card ui inconsistencies (backport #17160) (#17226)

fix: number card ui inconsistencies (#17160)

(cherry picked from commit 335edc14f3b66cd34c97deab0e0f807d71afe723)

Co-authored-by: Ritwik Puri <ritwikpuri5678@gmail.com>

hrwX committed:
feat: set image as footer in letter head (#17119)

* feat: set image as footer in letter head

* fix: do not hide footer section

* fix: hide footer source

* style: black

* fix: reduce code for setting image as html

* fix: Force system admin role only if active

* fix(ui): tab refresh was not implemented

* fix(minor): Onboarding: add option to view list view in create action

* fix(minor): js lint

* fix: allow All to select a User

* test: user permissions affecting User

* refactor: filter_dynamic_link_doctypes API

* Added typing, better variable naming
* Remove unnecessary re-iterations
* Optimize queries and membership processing

* perf: Check query type via is_query_type

* chore: Drop duplicate get_frontmatter definition

* perf: Login Page

Improves performance 3x - from 0.047s to 0.017s

* Use frappe.get_*_settings to query table once
* Use cached LDAP Settings' document via get_ldap_client_settings
* Use single get_all to query all Social Login providers and related
  data
* Skip provider if client_secret doesn't exist

* perf: About Us Settings

Use cached document for building /about page

* perf: App Page

Reduced time taken for get_context to execute from 0.035s to 0.02s (75%
reduction)

* perf: Patch qb only once - not on every init

* perf: Fetch and cache entire settings' dicts

* refactor!: frappe.db.get_singles_dict

* Cast single's values as their fieldtypes before returning
* Support previously dead debug parameter
* Consider single with no meta as non-existent; skip query

Decided to go ahead with the breaking change given the nature of the
existing usages of get_singles_dict :crie:

* ci: Run tests bypassing roulette with labels "Run UI Tests", "Run Server Tests"

* test: Scheduler tests cleanup

* chore: linter changes

* chore: linter changes

* chore: linter changes

* fix: set HTML as default

to keep existing formats working

* refactor: no **kwargs

Co-authored-by: Raffael Meyer <14891507+barredterra@users.noreply.github.com>
Co-authored-by: Deepesh Garg <deepeshgarg6@gmail.com>
Co-authored-by: Rushabh Mehta <rmehta@gmail.com>
Co-authored-by: Gavin D'souza <gavin18d@gmail.com>
(cherry picked from commit 6d6a67e9e322abd831d9669dc93c54c0b14dd833)

# Conflicts:
#	frappe/printing/doctype/letter_head/letter_head.json
#	frappe/printing/doctype/letter_head/letter_head.py

mergify[bot] committed:
fix: translate doctype in error messages (backport #17239) (#17240)

* fix: translate doctype in error messages (#17239)

(cherry picked from commit 7b67e1f847121f255c3262811bc54493a2563342)

# Conflicts:
#	frappe/__init__.py
#	frappe/client.py

* chore: conflicts

Co-authored-by: Raffael Meyer <14891507+barredterra@users.noreply.github.com>
Co-authored-by: Ankush Menat <ankush@frappe.io>

hrwX committed:
Update letter_head.json

hrwX committed:
Update letter_head.py

surajshetty3416 committed:
refactor: Replace usage of deprecated attribute

(cherry picked from commit 64cc07227eaac493865a851bde90ca5ca0480b7d)

surajshetty3416 committed:
refactor: Rename icon symbol filename from symbol-defs.svg to icons.svg

(cherry picked from commit 245302dc7a28aaf286f2798522cb707ba32879ec)

surajshetty3416 committed:
fix: Undo `html_utils.py` changes

surajshetty3416 committed:
Merge pull request #17255 from frappe/mergify/bp/version-13-hotfix/pr-17254

deepeshgarg007 committed:
Merge pull request #17256 from frappe/version-13-hotfix

chore: weekly version-13 release

frappe-bot committed:
chore(release): Bumped to Version 13.33.1

## [13.33.1](https://github.com/frappe/frappe/compare/v13.33.0...v13.33.1) (2022-06-21)

### Bug Fixes

* Add image_field check ([e717744](https://github.com/frappe/frappe/commit/e7177440454f2ffaf9238dd3f3f7f473e116d4e0))
* allow All to select a User (backport [#16983](https://github.com/frappe/frappe/issues/16983)) ([#17178](https://github.com/frappe/frappe/issues/17178)) ([210ad01](https://github.com/frappe/frappe/commit/210ad01177a5b90ec8ce9e8500b66c9a4748368f))
* **db_query:** Remove wanton whitespace split check (backport [#17180](https://github.com/frappe/frappe/issues/17180)) ([#17195](https://github.com/frappe/frappe/issues/17195)) ([df1c8af](https://github.com/frappe/frappe/commit/df1c8af240843e89917eae05e8f38a2eded26246))
* **Event Streaming:** set remote docname and site name before inserting mapped doc ([bf66f4e](https://github.com/frappe/frappe/commit/bf66f4e4fe92f7373c359e1b31c26b82a08a8e18))
* format date as per system settings for dashboard charts (backport [#17151](https://github.com/frappe/frappe/issues/17151)) ([#17203](https://github.com/frappe/frappe/issues/17203)) ([0fd7cba](https://github.com/frappe/frappe/commit/0fd7cbaef36a88b5871fdd12f3a0210f481fe2e2))
* Make images in comments private ([3e3b03e](https://github.com/frappe/frappe/commit/3e3b03e8fe8acf712dd734a9f5274b301f274a60))
* no add/change image-field if user is not allowed ([9ba1feb](https://github.com/frappe/frappe/commit/9ba1febbfe2e815e1f4f4c8c60df189ede4f0f04))
* number card ui inconsistencies (backport [#17160](https://github.com/frappe/frappe/issues/17160)) ([#17226](https://github.com/frappe/frappe/issues/17226)) ([c540940](https://github.com/frappe/frappe/commit/c5409401c16720ee827e0752ac480f9763fd0e33))
* Optional parameter for setting extracted image as private ([4e0fd17](https://github.com/frappe/frappe/commit/4e0fd177b5406d822d96ea772876b3c686880f35))
* Private images in comments are not visible ([cfca208](https://github.com/frappe/frappe/commit/cfca2085d2e90c61e5229b43608cad2be8504aab))
* Resolve conflicts ([dc2ddf9](https://github.com/frappe/frappe/commit/dc2ddf9a6cb6228d56357b27d7b22d10ba0bdb0c))
* translate doctype in error messages (backport [#17239](https://github.com/frappe/frappe/issues/17239)) ([#17240](https://github.com/frappe/frappe/issues/17240)) ([69612f7](https://github.com/frappe/frappe/commit/69612f75ee4564c3e0ab9ff94629eaa3b6bb14e7))
* Undo `html_utils.py` changes ([1d8383c](https://github.com/frappe/frappe/commit/1d8383c96ec3fd4d9ffd3251ca8f4b75cc42f72b))
* Update label for editing in full page and add custom logic for rendering it. ([#17149](https://github.com/frappe/frappe/issues/17149)) ([b321234](https://github.com/frappe/frappe/commit/b321234a41726b4044b86e407e62e86e5e0fded6))
* use the correct dt name while populating data in mapping table ([bb36bba](https://github.com/frappe/frappe/commit/bb36bbaf858d9e977494ff181857ffc1cefd940c))

### Reverts

* Revert "fix: allow All to select a User (backport #16983)" (#17199) ([5288229](https://github.com/frappe/frappe/commit/5288229aa55531af239fbcd83565240f3ba9bc61)), closes [#16983](https://github.com/frappe/frappe/issues/16983) [#17199](https://github.com/frappe/frappe/issues/17199) [#16983](https://github.com/frappe/frappe/issues/16983) [#17178](https://github.com/frappe/frappe/issues/17178)

Commit:
Merge remote-tracking branch 'upstream/version-13' into merge-v13

scdanieli committed:
fix(Image View): translate info field

(cherry picked from commit 3bc636ce30ba702a77c4d1e77a0fb46ffeed2e83)

ankush committed:
perf: improve Document caching (#17301)

Partial port of perf fix from https://github.com/frappe/frappe/pull/17061

hrwX committed:
Merge branch 'version-13-hotfix' into mergify/bp/version-13-hotfix/pr-17119

hrwX committed:
Update letter_head.py

mergify[bot] committed:
fix: Allow permitted number card of type report (backport #17316) (#17319)

Co-authored-by: Shariq Ansari <30859809+shariquerik@users.noreply.github.com>

surajshetty3416 committed:
Merge pull request #17229 from frappe/mergify/bp/version-13-hotfix/pr-17119

surajshetty3416 committed:
Merge pull request #17291 from frappe/mergify/bp/version-13-hotfix/pr-17287

fix(Image View): make info field translatable (backport #17287)

deepeshgarg007 committed:
Merge pull request #17326 from frappe/version-13-hotfix

chore: weekly version-13 release

frappe-bot committed:
chore(release): Bumped to Version 13.34.0

# [13.34.0](https://github.com/frappe/frappe/compare/v13.33.1...v13.34.0) (2022-06-28)

### Bug Fixes

* Allow permitted number card of type report (backport [#17316](https://github.com/frappe/frappe/issues/17316)) ([#17319](https://github.com/frappe/frappe/issues/17319)) ([de2ec9b](https://github.com/frappe/frappe/commit/de2ec9ba755a0773bb782211aa83cb815b1d59e6))
* **Image View:** translate info field ([0eab450](https://github.com/frappe/frappe/commit/0eab450daee87b0cd9759fff8ad094130f430fde))

### Features

* set image as footer in letter head ([#17119](https://github.com/frappe/frappe/issues/17119)) ([fa0376f](https://github.com/frappe/frappe/commit/fa0376f696e18779eba69c9c04fdbb23934a0d45))

### Performance Improvements

* improve Document caching ([#17301](https://github.com/frappe/frappe/issues/17301)) ([c21ab07](https://github.com/frappe/frappe/commit/c21ab0788897f77b1ae757f8afc2c1221d22e60b))

Commit:
Merge remote-tracking branch 'upstream/version-13' into merge-v13
